ocaml/typing/typecore.ml

```diff
@@ -3698,10 +3697,23 @@ and type_approx_aux env sexp in_function ty_expected =
   | None      -> match sexp.pexp_desc with
     Pexp_let (_, _, e) -> type_approx_aux env e None ty_expected
   | Pexp_fun (l, _, p, e) ->
+      let l, inner_pat = match p with
+      | {ppat_desc = Ppat_constraint (inner_pat, ty); _} ->
+          (* If the argument is a constraint, translate the label using the
+             type information. Otherwise, it can't be a Position argument, so
+             we don't care about the argument type *)
+          Typetexp.transl_label l (Some ty), inner_pat
+      | _ -> Typetexp.transl_label l None, p
+      in
+      let p = if is_position l then inner_pat else p in
+      (* TODO vding: This code is still kind of ugly because now, [inner_pat]
+         is sometimes something inside the pattern, sometimes the pattern 
+         itself, and sometimes we want to use it, sometimes we don't. Maybe
+         it's not that bad? *)
```

Comment:
^

Reply:
I think a good approach here would be to have some variant of `transl_label` (maybe `transl_label_from_pat` that returns the `Types.arg_label` along with the pattern to check. It would have to take a pattern as an argument and look for the constraint.

Reply:
Implemented and I put it in `Typetexp`, next to `transl_label`. 
Is it weird here since it's only used in `Typecore`, and doesn't really have anything to do with translating type expressions? Or is it nice to have it next to `transl_label`

Reply:
I would make the same call as you did: in `Typetexp`. It just kinda belongs there.